### PR TITLE
UI update

### PR DESCRIPTION
**Pull request by mikedzikowski**
Updates for UI that add optional resource group creation (new or existing)

### CODE REVIEW

File: uiDefinition.json [hunk 1]
```diff
@@ -18,37 +18,76 @@
...
+                        "visible": false,
+                        "type": "Microsoft.Solutions.ArmApiControl",
+                        "defaultValue": "",
+                        "request": {
+                            "method": "GET",
+                            "path": "[concat(steps('basics').scope.subscription.id, '/resourceGroups?api-version=2021-04-01')]"
+                        }
+                    },
+                    {
+                        "name": "newOrExistingResourceGroup",
```
Comment (jamasten) [inline uiDefinition.json]: Remove "newOr" from input name

File: uiDefinition.json [hunk 2]
```diff
@@ -18,37 +18,76 @@
...
-                            "constraints": {
-                                "allowedValues": "[map(steps('basics').resourceGroupsApi.value, (item) => parse(concat('{\"label\":\"', item.name, '\",\"value\":\"', item.name, '\"}')))]",
-                                "required": true
-                            },
-                            "infoMessages": [],
-                            "visible": true
+                        "visible": true
+                    },
+                    {
+                        "name": "existingResourceGroup",
```
Comment (jamasten) [inline uiDefinition.json]: Change to "existingResourceGroupName"

Comment (mikedzikowski) [reply uiDefinition.json]: updated UI definition based on peer review

File: uiDefinition.json (hunk 1 above)
Comment (mikedzikowski) [reply uiDefinition.json]: updated UI definition based on peer review

**Review by jamasten — APPROVED** [review]

**Pull request merged**